examples/datasets/plot_limo_data.py

```diff
@@ -236,63 +215,52 @@
-# and occipital electrodes.
-
-###############################################################################
 # Prepare data for linear regression analysis
 # --------------------------------------------
 #
-# Next, we can test the significance of these differences using linear
-# regression.But before we go on, we'll start by interpolating any missing
-# channels in the LIMO epochs structure. Some subjects in the datasets contain
-# missing channels (stored in ``limo_epochs.info[‘bads’]``) as these were
+# Before we test the significance of these differences using linear
+# regression, we'll interpolate missing channels that were
 # dropped during preprocessing of the data.
+# Furthermore, we'll drop the EOG channels (marked by the "EXG" prefix)
+# present in the data:
 
 limo_epochs.interpolate_bads(reset_bads=True)
 
-# Furthermore, we'll drop the EOG channels (marked by the "EXG" prefix)
-# present in the LIMO epochs structure.
 limo_epochs.drop_channels(['EXG1', 'EXG2', 'EXG3', 'EXG4'])
 
 ###############################################################################
 # Define predictor variables and design matrix
 # --------------------------------------------
 #
-# First, we need to create a design matrix containing information about the
+# To run the regression analysis,
+# we need to create a design matrix containing information about the
 # variables (i.e., predictors) we want to use for prediction of brain
-# activity patterns. For this purpose, we'll use the information contained in
-# the ``limo_epochs.metadata``. Here, we'll explore the effect of
-# phase-coherence as well as the effect of Face A vs. Face B.
+# activity patterns. For this purpose, we'll use the information we have in
+# ``limo_epochs.metadata``: phase-coherence and Face A vs. Face B.
 
 # name of predictors + intercept
 predictor_vars = ['face a - face b', 'phase-coherence', 'intercept']
 
 # create design matrix
-design = limo_epochs.metadata[['phase-coherence', 'face']]
+design = limo_epochs.metadata[['phase-coherence', 'face']].copy()
 design['face a - face b'] = np.where(design['face'] == 'A', 1, -1)
 design['intercept'] = 1
 design = design[predictor_vars]
 
 ###############################################################################
 # Now we can set up the linear model to be used in the analysis.
-# For the purpose of demonstration, we'll use scikit-learn's LinearRegression
-# (see :class:`sklearn.linear_model.LinearRegression`) in this example.
-#
-# Basically, we're creating a wrapper, ``STLinearRegression``, for
-# sklearn's ``LinearRegression``, which computes a
-# `least squares`_ solution for our data given the provided design matrix. The
-# and results are stored within ``LinearRegression`` object for convenience.
+# We'll use :class:`sklearn.linear_model.LinearRegression`, and create a
+# wrapper ``STLinearRegression`` that computes a
+# `least squares`_ solution for our data given the provided design matrix.
```

Comment:
I actually think you should remove the entire `sklearn` model.  If the MNE-Python regression gives the same results (and can be done in just a few lines), I think the added complexity of the wrapper class, etc, just distracts from the main point of the tutorial.  I'm happy to keep it in if you think the `sklearn` approach adds something important to the tutorial, but you'd have to convince me.

At the very least, I'd put the mne-python regression first, so people can see the results right away.  Then, if you want to show something more complex that only `sklearn` can do, then move to the more complex `sklearn` model afterward.

Reply:
See discussion on the main PR, hope the arguments are somewhat convincing :)